apollo-federation/src/query_graph/mod.rs

```diff
@@ -901,17 +899,15 @@ impl QueryGraph {
             .federation_spec_definition()
             .provides_directive_definition(schema)?;
 
-        for object_type_definition_position in
-            schema.possible_runtime_types(interface_field_definition_position.parent().into())?
-        {
-            let field_pos = object_type_definition_position
-                .field(interface_field_definition_position.field_name.clone());
-            let field = field_pos.get(schema.schema())?;
-            if field.directives.has(&provides_directive_definition.name) {
-                return Ok(true);
-            }
-        }
-
-        Ok(false)
+        schema
+            .possible_runtime_types(interface_field_definition_position.parent().into())?
+            .into_iter()
+            .map(|object_type_definition_position| {
+                object_type_definition_position
+                    .field(interface_field_definition_position.field_name.clone())
+            })
+            .map(|field_pos| field_pos.get(schema.schema()))
+            .ok_and_any(|field| field.directives.has(&provides_directive_definition.name))
+            .map_err(Into::into)
```

Comment:
Is there a nice way to replace this map_err with a `?` operator?

Reply:
`map_err(Into::into)` can be replaced with `?` and wrapping the whole express in an `Ok`. I generally dislike `Ok(fallible()?)`, but that's kind of bikeshedding. It doesn't really matter.